inventory/group_by.py

```diff
@@ -30,6 +30,9 @@
 author: "Jeroen Hoekx (@jhoekx)"
 notes:
   - Spaces in group names are converted to dashes '-'.
+  - Use of 'when' directive cause unpredictable results.
```

Comment:
I'm not a native speaker, but I think the message should be improved.

Something like: 
- Use of group_by with the when statement causes unpredictable results.
- Only variables of the first host in the inventory are available.